projectq/ops/_command.py

```diff
@@ -152,6 +151,24 @@ def is_identity(self):
             True if the gate is equivalent to an Identity gate, False otherwise
         """
         return projectq.ops.is_identity(self.gate)
+    
+    def is_commutable(self, other):
+        """
+        Evaluate if this command is commutable with another command.
+        Returns:
+            True if the other command concerns the same qubits and
+            the gates are commutable. 
+        """
+        if (overlap(self.all_qubits, other.all_qubits)==0):
+            return 0
+        self._commutable_circuit_list = self.gate.get_commutable_circuit_list(len(self.control_qubits))
+        # If other gate may be part of a list which is 
+        # commutable with gate, return 2
+        for circuit in self._commutable_circuit_list:
+            if (other.gate.__class__ == circuit[0]._gate.__class__):
+                return 2
+        else:
+            return self.gate.is_commutable(other.gate)
```

Comment:
I have changed is_commutable to work with enums. . is_commutable now returns a value such as: Commutable.COMMUTABLE.value. I chose to keep it returning an integer (the value of COMMUTABLE) rather than an enum, because then in my test files I can keep the nice simple readability of ‘assert gate1.is_commutable(gate2)’ and ‘assert not gate2.is_commutable(gate3)’. For the third option (a commutable circuit) you use: ‘assert gate3.is_commutable(gate4) == Commutable.MAYBE_COMMUTABLE

So this method already returns an int

Reply:
See comment https://github.com/ProjectQ-Framework/ProjectQ/pull/386#discussion_r607637535

Reply:
Done. As will be seen in next commit.